### ISSUE TITLE
Implement API Integration

### ISSUE THREAD

**Issue by kaybrian** [pycon-rwanda/Endangered-Species-Tracker#1]
- Write code to connect to the IUCN Red List API and fetch data about endangered species. Ensure to handle API keys securely.

**Comment by JohananOppongAmoateng** [comment]
@kaybrian assign this to me

**Comment by kaybrian** [comment]
Got  you on that.

**Comment by JohananOppongAmoateng** [comment]
@kaybrian the version 3 of the api is being retired
Not seeing any endpoint for using species name to fetch data

**Comment by kaybrian** [comment]
> @kaybrian the version 3 of the api is being retired Not seeing any endpoint for using species name to fetch data

I have been able to look at the new API, I think we can currently use API version 4 since its out there and active 

here is the URL -> https://api.iucnredlist.org/api-docs/index.html

it can be of help

**Comment by kaybrian** [comment]
I have also been able to write some fetch functions, let me make a pr so you look into it

**Comment by kaybrian** [comment]
@JohananOppongAmoateng are you ok with this issue?

**Comment by JohananOppongAmoateng** [comment]
@kaybrian I think it's been worked on. Are there any new things you want added?

**Comment by kaybrian** [comment]
Yes. When you look at the information and api calls we have, I think we need more things

**Comment by JohananOppongAmoateng** [comment]
Can you specify what we need so I work on it?
